packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,277 @@
+      return;
+    }
+
+    // If needed, set default content sensitivity level as set in native Android. This will be
+    // auto sensitive if it is otherwise unset by the developer. Also, initialize the current
+    // content sensitivity level if needed.
+    _defaultContentSensitivitySetting ??= await _sensitiveContentService.getContentSensitivity();
+    currentContentSensitivityLevel ??= _defaultContentSensitivitySetting;
+
+    // Update SensitiveContent widget count for those with desiredSensitivityLevel.
+    _contentSensitivitySetting.addWidgetWithContentSensitivity(desiredSensitivityLevel);
+
+    // Verify that desiredSensitivityLevel should be set in order for sensitive
+    // content to remain obscured.
+    if (!shouldSetContentSensitivity(desiredSensitivityLevel)) {
+      return;
+    }
+
+    // Set content sensitivity level as desiredSensitivityLevel. If the call to set content
+    // sensitivity on the platform side fails, then we do not update the current content
+    // sensitivity level.
+    _sensitiveContentService.setContentSensitivity(desiredSensitivityLevel);
+
+    // Update current content sensitivity level.
+    currentContentSensitivityLevel = desiredSensitivityLevel;
+  }
+
+  /// Unregisters a [SensitiveContent] widget from the [ContentSensitivitySetting] tracking
+  /// the content sensitivity level of the widget tree.
+  static void unregister(ContentSensitivity widgetSensitivityLevel) {
+    instance._unregister(widgetSensitivityLevel);
+  }
+
+  void _unregister(ContentSensitivity widgetSensitivityLevel) {
+    assert(
+      _contentSenstivityIsSupported != null,
+      'SensitiveContentHost.register must be called before SensitiveContentHost.unregister',
+    );
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // Update SensitiveContent widget count for those with
+    // desiredSensitivityLevel.
+    _contentSensitivitySetting.removeWidgetWithContentSensitivity(widgetSensitivityLevel);
+
+    if (!_contentSensitivitySetting.hasWidgets) {
+      // Restore default content sensitivity setting if there are no more SensitiveContent
+      // widgets in the tree. If the call to set content sensitivity on the platform side fails,
+      // then we do not update the current content sensitivity level.
+      _sensitiveContentService.setContentSensitivity(_defaultContentSensitivitySetting!);
+
+      // Update current content sensitivity level.
+      currentContentSensitivityLevel = _defaultContentSensitivitySetting;
+      return;
+    }
+
+    // Determine if another sensitivity level needs to be restored.
+    late final ContentSensitivity? contentSensitivityToRestore =
+        _contentSensitivitySetting.contentSensitivityBasedOnWidgetCounts;
+    if (contentSensitivityToRestore != null &&
+        contentSensitivityToRestore != currentContentSensitivityLevel) {
```

Comment:
I am not sure this check helps with efficiency because the java code will only set the value if it needs to be.

Reply:
But then we are sending messages across the method channel for no reason